conda/core/solve.py

```diff
@@ -607,12 +626,27 @@ def _add_specs(self, ssc):
                                       for prec in ssc.prefix_data.iter_records()
                                       )
 
+        if ssc.update_modifier == UpdateModifier.UPDATE_SPECS:
+            for spec in self.specs_to_add:
+                spec = MatchSpec(spec)
+                if (spec.name in pin_overrides and not ssc.ignore_pinned
+                        or spec.name in ssc.specs_from_history_map):
+                    # skip this spec, because it is constrained by pins
+                    continue
+                # the index is sorted, so the first record here gives us what we want.
+                latest_pkg = ssc.r.find_matches(spec)[0]
```

Comment:
There is no certainty that the first record here is what will be or even should be installed.  If a particular variant sorts before the variant that will be installed then many specs are un-necessarily neutered.  For example in an environment with openssl 1.1.1 if a package has an openssl 1.1.1 and 1.0.2 variant if the 1.0.2 variant had sorted before the 1.1.1 variant then openssl would neutered for no reason.

Reply:
I'm not following why this section is needed at all.  Is there something in the existing neutering logic that was causing issues?

Reply:
The first record is a statement of the newest version available.  It doesn't say anything about satisfiability.  It only relaxes (trivially found) conflicts such that the newest version _might_ be installable.  I think this is OK.

Reply:
Right, I'm following this now.  Rhis happens before the explicitly request specs are added (around line 680) so there is no chance that they will be neutered.